packages/plugin-core/src/test/suite-integ/AddAndCommitCommand.test.ts

```diff
@@ -0,0 +1,215 @@
+import { ConfigUtils, NoteUtils } from "@dendronhq/common-all";
+import { tmpDir } from "@dendronhq/common-server";
+import {
+  Git,
+  SyncActionStatus,
+  WorkspaceUtils,
+} from "@dendronhq/engine-server";
+import { GitTestUtils } from "@dendronhq/engine-test-utils";
+import _ from "lodash";
+import { expect } from "../testUtilsv2";
+import { describeSingleWS } from "../testUtilsV3";
+import fs from "fs-extra";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { AddAndCommit } from "../../commands/AddAndCommit";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+
+const TIMEOUT = 60 * 1000 * 5;
+
+suite("GIVEN Workspace Add And Commit command is run", function () {
+  this.timeout(TIMEOUT);
+  describeSingleWS(
+    "WHEN there are no changes",
+    {
+      modConfigCb: (config) => {
+        ConfigUtils.setWorkspaceProp(config, "workspaceVaultSyncMode", "sync");
+        return config;
+      },
+    },
+    () => {
+      test("THEN Dendron skips committing", async () => {
+        const { wsRoot } = ExtensionProvider.getDWorkspace();
```

Comment:
the code for creating a git remote seems generic and re-used in multiple functions, can we extract that as a utility instead?